prboom2/src/p_setup.c

```diff
@@ -3681,6 +3681,10 @@ void P_SetupLevel(int episode, int map, int playermask, int skill)
   {
     S_ParseMusInfo(lumpname);
```

Comment:
Could you instead migrate the condition here into `S_ParseMusInfo`: `if (gamemode != shareware && W_LumpNameExists("MUSINFO"))`? That function already has the else condition so it should isolate the behavior better.